cmd/nagios_enqueue.go

```diff
@@ -0,0 +1,157 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/PagerDuty/go-pdagent/pkg/eventsapi"
+	"github.com/spf13/cobra"
+)
+
+const defaultNagiosIntegrationSeverity = "error"
+
+var allowedNotificationTypes = []string{"PROBLEM", "ACKNOWLEDGEMENT", "RECOVERY"}
+var allowedSourceTypes = []string{"host", "service"}
+
+var errNotificationType = fmt.Errorf("notification-type must be one of: %v", strings.Join(allowedNotificationTypes, ", "))
+var errSourceType = fmt.Errorf("source-type must be one of: %v", strings.Join(allowedSourceTypes, ", "))
+
+var requiredFields = map[string][]string{
+	"host":    {"HOSTNAME", "HOSTSTATE"},
+	"service": {"HOSTNAME", "SERVICEDESC", "SERVICESTATE"},
+}
+
+var nagiosToPagerDutyEventType = map[string]string{
+	"PROBLEM":         "trigger",
+	"ACKNOWLEDGEMENT": "acknowledge",
+	"RECOVERY":        "resolve",
+}
+
+func NewNagiosEnqueueCmd(config *Config) *cobra.Command {
+	var customDetails map[string]string
+
+	var sourceType string
+
+	var sendEvent = eventsapi.EventV2{
+		Payload: eventsapi.PayloadV2{},
+	}
+
+	requiredFlags := []string{"routing-key", "notification-type", "source-type"}
+
+	cmd := &cobra.Command{
+		Use:   "enqueue",
+		Short: "Enqueue an event from Nagios to PagerDuty.",
+		Long: fmt.Sprintf(`Enqueue an event from Nagios to PagerDuty.
+
+	The following flags are required to be set for this command: %v.
+
+	When the source type is "host", the following fields must be set using the -f flag:
+	%v
+
+	When the source type is "service", the following fields must be set using the -f flag:
+	%v
+		`, strings.Join(requiredFlags, ", "), strings.Join(requiredFields["host"], ", "), strings.Join(requiredFields["service"], ", ")),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := validateNagiosSendCommand(sendEvent, sourceType, customDetails)
+			if err != nil {
+				return err
+			}
+
+			transformedSendEvent, transformedCustomDetails := nagiosTransformations(sendEvent, sourceType, customDetails)
+			return runSendCommand(config, transformedSendEvent, transformedCustomDetails)
+		},
+	}
+
+	cmd.Flags().StringVarP(&sendEvent.RoutingKey, "routing-key", "k", "", "Service Events API Key (required)")
+	cmd.Flags().StringVarP(&sendEvent.EventAction, "notification-type", "t", "", "The Nagios notification type (required)")
+	cmd.Flags().StringVarP(&sourceType, "source-type", "u", "", "The Nagios source type (host or service, required)")
```

Comment:
This is my fault.  I apologize in advance.

Could we change this back to `-n` to match what we previously had?

Reply:
Updated here: 1d16759